src/vm.rs

```diff
@@ -419,6 +424,60 @@ fn matches_literal(s: &str, ix: usize, end: usize, literal: &str) -> bool {
     end <= s.len() && &s.as_bytes()[ix..end] == literal.as_bytes()
 }
 
+#[inline]
```

Comment:
I'd probably not force inlining of this as it's pretty long (and probably not used often). AFAIK leaving it off will mean that the compiler will decide.

Reply:
good point, thanks, fixed :)